hot/template-loader.js

```diff
@@ -14,13 +14,25 @@ module.exports.pitch = function (request) {
     return `module.exports = require(${moduleId});`;
   }
 
+  // hot reloading of templates works by exporting a wrapped closure function that references templateFn
+  // when we receive a new module update, we swap templateFn reference to new template function
+  // this means components that use the template don't need to change their function reference
+  // we just need to call their .update()
   return `
-    let template = require(${moduleId});
+    const getTemplateFn = (mod) => mod.__esModule ? mod.template : mod;
+    const templateModule = require(${moduleId});
+    let templateFn = getTemplateFn(templateModule);
+    const wrappedTemplateFn = function() {return templateFn.apply(this, arguments)};
     module.hot.accept(${moduleId}, () => {
-      template = require(${moduleId});
+      const templateModule = require(${moduleId});
```

Comment:
I don't really like the `templateModule` shadowing here but it is not a deal-breaker. We could inline the call to avoid it.

```ts
templateFn = getTemplateFn(require(${moduleId}))
```

Reply:
Makes sense. Eslint would have yelled. Will change.